File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        num_of_str = len(strs)

        if num_of_str == 1:
            return strs[0]
        
        result = []
        j = 0
        for ch in strs[0]:
            i = 1
            while i  < num_of_str and j < len(strs[i]):
                if ch == strs[i][j]:
                    i+=1          
                else:
                    return "".join(result)
                
                if i == num_of_str:
                    result.append(ch)
                    j+=1

        return "".join(result)

File: 14-longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if(strs.length === 0) return "";
    
    let i = 0;
    let prefix = "";
    let minLength = Infinity;


    for(const s of strs){
        if(s.length < minLength){
            minLength = s.length;
        }
    }

    while(i < minLength){
        for(const s of strs){
            if(s[i] !== strs[0][i]){
                return s.slice(0,i)
            }
        }
        i+=1;
    }

    return strs[0].slice(0,i);

};